rfcs/0005-fabric-elements-api.md

```diff
@@ -0,0 +1,43 @@
+# RFC: Fabric Elements API
+
+## Background
+
+When building our Web Component library (Fabric Elements) it's good to think of the overall picture of how the API should be structured. 
+With the right conventions, it would be easier to maintain consistency when adding new components, which should result in the library being easy to work with for both authors and users.
+
+This RFC attempts to start a discussion on how to structure the Fabric Elements API in a way that aims to minimise confusion and complexity of various components.
+
+## Concepts/Principles to discuss
+
+### Customization of components
+
+To what degree should web components be customized with respective attributes if they return just one main HTML element,
+e.g. when `<f-textarea>` returns `<textarea>`, `<f-button>` returns `<button>` or `<f-textfield>` returns `<input>`? 
+Should we bind all possible attributes to respective WC properties in case the user wants to change these?
+If we render multiple components on a page, each binding all possible attributes + being placed within shadow DOM, 
+would that affect page performance much?
+
+### Naming of component attributes
+
+In most web component libraries the component attributes are called the same as native element attributes, e.g. `disabled`, `target`, `aria-label`.
```

Comment:
From my limited experience with web components, I can only say that it looks like WC libs try to be as transparent as possible, mimicing the behaviour and attributes of native HTML elements. That could also be a good strategy for components in other frameworks, to expose eg. an `onChange` prop in a `Combobox` component where an `change` event would be expected in a conceptually related `select` element.